src/main/groovy/wooga/gradle/unity/utils/internal/UnityAssetFile.groovy

```diff
@@ -19,57 +19,147 @@ package wooga.gradle.unity.utils.internal
 
 import org.gradle.api.logging.Logger
 import org.gradle.api.logging.Logging
+import org.yaml.snakeyaml.DumperOptions
 import org.yaml.snakeyaml.Yaml
+import org.yaml.snakeyaml.nodes.Tag
+import org.yaml.snakeyaml.representer.Represent
+import org.yaml.snakeyaml.representer.Representer
+import org.yaml.snakeyaml.nodes.Node;
 
 abstract class UnityAssetFile {
     static Logger logger = Logging.getLogger(UnityAssetFile)
     private final validObject
+    private final File assetFile
 
     private Map<String, Object> content = [:]
+    private List<String> unityInstructions
 
     UnityAssetFile(File assetFile) {
         this(assetFile.text)
+        this.assetFile = assetFile
     }
 
     UnityAssetFile(String content) {
         boolean isValid = false
         try {
             Yaml parser = new Yaml()
-            Map<String, Object> c = parser.load(stripUnityInstructions(content))
+            def lines = content.readLines()
+            lines.removeAll({ it -> it.isEmpty()})
+            unityInstructions = lines[0,1,2]
+            lines[1] = ""
+            lines[2] = "--- "
+            String stripped = lines.join("\n")
+            Map<String, Object> c = parser.load(stripped)
```

Comment:
Is it always the first 3 lines with the special unity markup?

Reply:
Yes, outside of one edge case (XRSettings), they write the first 3 everytime.